students/elaine_x/session03/mailroom.py

```diff
@@ -0,0 +1,104 @@
+'''
+##########################
+#Python 210
+#Session 03 - Mailroom Part 1
+#Elaine Xu
+#Jan 29,2019
+###########################
+'''
+
+def menu():
+    '''print out the main menu of options'''
+    print('''Menu
+    Please choose from below options:
+    1 - Send a Thank You
+    2 - Create a Report
+    3 - quit''')
+
+
+def send_a_thankyou():
+    '''send a thankyou note to donor'''
+    # creating a list of donors
+    donor_list = []
+    for donor in donor_db:
+        donor_list.append(donor[0].lower())
+    #print(donor_list) #for testing
+    #prompt for a full name
+    while True:
+        fullname = input("Enter full name of the donor (type 'list' to show all donor names): ")
+        if fullname.lower() == 'list':
+            for donor in donor_db:
+                print(donor[0])
+        elif fullname.lower() not in donor_list:
+            print(f'{fullname} does not exist in current donor data, '
+                  f'adding {fullname} to the donor data.')
+            #prompt for a donation amount
+            donation_amount = float(input("Enter the donation amount: "))
+            donor_db.append((fullname, [donation_amount, 1]))
+            print(f"{donation_amount} has been added to {fullname}'s donation history.")
+            #print(donor_db) #for testing
+            break
+        elif fullname.lower() in donor_list:
+            fullname_index = donor_list.index(fullname.lower())
+            #prompt for a donation amount
+            donation_amount = float(input("Enter the donation amount: "))
+            donor_db[fullname_index][1][0] = donor_db[fullname_index][1][0] + donation_amount
+            donor_db[fullname_index][1][1] = donor_db[fullname_index][1][1] + 1
+            print(f"{donation_amount} has been added to {fullname}'s donation history.")
+            #print(donor_db) #for testing
+            break
+    #send thankyou note
+    print("Composing Thank You email:")
+    print(f'Thank you {fullname} for your generous donation of {donation_amount:^10.2f}!')
+    print()
+    menu()
```

Comment:
You don't want to call the `menu()` function here as you'll get into a weird recursion thing when trying to exit.  You can just leave this line out and it will return to your `while True` loop for your menu.

Reply:
nvm, don't think this will do recursion since its only a text print out